src/DebuggerData.cpp

```diff
@@ -314,13 +319,13 @@ bool Breakpoints::isWatchpoint(quint16 addr, QString *id)
 
 int Breakpoints::findBreakpoint(quint16 addr)
 {
-    auto i = std::lower_bound(breakpoints.begin(), breakpoints.end(), addr,
+	auto i = std::lower_bound(breakpoints.begin(), breakpoints.end(), addr,
 	           [](const Breakpoint& bp, const quint16 addr) {
 				   return bp.address < addr;
 			   }
    			);
 	if (i != breakpoints.end() && i->address == addr) {
-		return i - breakpoints.begin();
+		return std::distance(breakpoints.begin(), i);
```

Comment:
This is almost of a philosophical remark. Feel free to completely ignore it.

Interesting change. I never really thought about the difference between `std::distance()` and `operator-` (on random access iterators) till now:
* `std::distance()` works on any type of iterator and automatically picks the most efficient implementation: O(1) for random-access-iterators, O(n) for forward-iterators, ...
* `operator-` only works on random-access-iterators, but is guaranteed to be O(1).

In this context you do a binary-search because you want a faster (in big-O terms) than linear algorithm. So a `std::distance()` that runs in O(n) would defeat that goal. Therefor, personally, I wouldn't mind to get a compile error if `operator-` on the iterators doesn't work anymore (e.g. because of some future refactoring that changes the container type of `breakpoints`).